aten/src/ATen/xpu/profiler/XPUActivityProfiler.cpp

```diff
@@ -0,0 +1,178 @@
+  profilerStartTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::stop() {
+  xpti_.disablePtiActivities(activity_types_);
+  profilerEndTs_ =
+      libkineto::timeSinceEpoch(std::chrono::high_resolution_clock::now());
+}
+
+void XPUActivityProfilerSession::processTrace(logger_t& logger) {
+  traceBuffer_.span = libkineto::TraceSpan(
+      profilerStartTs_, profilerEndTs_, "__xpu_profiler__");
+  traceBuffer_.span.iteration = iterationCount_++;
+  auto gpuBuffer = xpti_.activityBuffers();
+  if (gpuBuffer) {
+    xpti_.processActivities(
+        *gpuBuffer,
+        std::bind(
+            &XPUActivityProfilerSession::handlePtiActivity,
+            this,
+            std::placeholders::_1,
+            &logger));
+  }
+}
+
+void XPUActivityProfilerSession::processTrace(
+    logger_t& logger,
+    libkineto::getLinkedActivityCallback get_linked_activity,
+    int64_t captureWindowStartTime,
+    int64_t captureWindowEndTime) {
+  captureWindowStartTime_ = captureWindowStartTime;
+  captureWindowEndTime_ = captureWindowEndTime;
+  cpuActivity_ = get_linked_activity;
+  processTrace(logger);
+}
+
+std::unique_ptr<libkineto::DeviceInfo> XPUActivityProfilerSession::
+    getDeviceInfo() {
+  return {};
+}
+
+std::vector<libkineto::ResourceInfo> XPUActivityProfilerSession::
+    getResourceInfos() {
+  return {};
+}
+
+std::unique_ptr<libkineto::CpuTraceBuffer> XPUActivityProfilerSession::
+    getTraceBuffer() {
+  return std::make_unique<libkineto::CpuTraceBuffer>(std::move(traceBuffer_));
+}
+
+void XPUActivityProfilerSession::pushCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::popCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::Default);
+}
+
+void XPUActivityProfilerSession::pushUserCorrelationId(uint64_t id) {
+  xpti_.pushCorrelationID(id, XPUActivityApi::CorrelationFlowType::User);
+}
+
+void XPUActivityProfilerSession::popUserCorrelationId() {
+  xpti_.popCorrelationID(XPUActivityApi::CorrelationFlowType::User);
+}
+
+void XPUActivityProfilerSession::enumDeviceUUIDs() {
+  auto platform_list = sycl::platform::get_platforms();
+  // Enumerated GPU devices from the specific platform.
+  for (const auto& platform : platform_list) {
+    if (platform.get_backend() != sycl::backend::ext_oneapi_level_zero) {
+      continue;
+    }
+    auto device_list = platform.get_devices();
+    for (const auto& device : device_list) {
+      if (device.is_gpu()) {
+        if (device.has(sycl::aspect::ext_intel_device_info_uuid)) {
+          deviceUUIDs_.push_back(
+              device.get_info<sycl::ext::intel::info::device::uuid>());
+        } else {
+          std::cerr << "UUID is not supported for this XPU device" << std::endl;
+          deviceUUIDs_.push_back(std::array<unsigned char, 16>{});
+        }
+      }
+    }
+  }
```

Comment:
```suggestion
  for (const auto i : c10::irange(c10::xpu::device_count())) {
    auto& device = c10::xpu::get_raw_device(i);
    TORCH_CHECK(device.has(sycl::aspect::ext_intel_device_info_uuid));
    deviceUUIDs_.push_back(
              device.get_info<sycl::ext::intel::info::device::uuid>());
  }
```
If no uuid, we should abort immetiately.

Reply:
One more thing, If `XPUActivityProfilerSession` is not a singleton, you should cache `deviceUUIDs_` instead of fetching them in the construction. Because the SYCL runtime overhead is heavy.

Reply:
Let's check what kind of SYCL device does not have UUID first.
Even without UUID, it's not a good idea to assert the application.
We'd better warn user about this, like tell, no UUID, all devices will be marked as single device 0.

Reply:
> If no uuid, we should abort immetiately.

We can not use c10 because xpu kineto needs to decouple with anything from torch.

Reply:
> One more thing, If `XPUActivityProfilerSession` is not a singleton, you should cache `deviceUUIDs_` instead of fetching them in the constructor. Because the SYCL runtime overhead is heavy.

It is not a singleton. But it is usually instantiated once in workload.
I can make `deviceUUIDs_` to be static member to avoid redundant created.

Reply:
> Let's check what kind of SYCL device does not have UUID first. Even without UUID, it's not a good idea to assert the application. We'd better warn user about this, like tell, no UUID, all devices will be marked as single device 0.

Are there any sycl guys know which platform doesn't support UUIDs? I try to find who enables the UUIDs feature in llvm/sycl.

Reply:
https://github.com/intel/llvm/blob/4cb0518de717b1a8d45d9ea8b94a077062b9d331/sycl/doc/extensions/supported/sycl_ext_intel_device_info.md?plain=1#L61

Reply:
If there is no uuid, could you use the other unique value to replace uuid?

Reply:
> If there is no uuid, could you use the other unique value to replace uuid?

If there is no UUID support on this platform, that means the record, returned by the PTI, also has no UUID info, then we cannot identify the device even if we have self-defined UUIDs because the UUID in all records is 0. Here we plan to throw a warning and set all device to be 0 in kineto.

Reply:
Is it reasonable to sync with pti that returns `std::hash` that is unique if there is no uuid.

Reply:
> Is it reasonable to sync with pti that returns `std::hash` that is unique if there is no uuid.

The record's device UUID is also get from the level zero. PTI doesn't know which device the kernel in this record is running on without UUID.

Reply:
I see. Thanks.

Reply:
ok